vllm/config.py

```diff
@@ -276,12 +274,6 @@ def verify_with_parallel_config(
                 "Pipeline parallelism is only supported for the following "
                 f" architectures: {_PP_SUPPORTED_MODELS}.")
 
-        if total_num_hidden_layers % pipeline_parallel_size != 0:
```

Comment:
Shouldn't we error out when `total_num_hidden_layers < pipeline_parallel_size `?

Reply:
speculative decoding has `total_num_hidden_layers=0` for some statistical worker (e.g. ngram).